app/src/main/java/com/example/bookwormadventuresdeluxe2/FirebaseUserGetSet.java

```diff
@@ -0,0 +1,186 @@
+    }
+
+    /**
+     * Callback for UserProfileObject
+     */
+    public interface UserCallback
+    /*
+    * Source: https://stackoverflow.com/questions/49514859/how-to-get-data-object-from-another-event-android-studio
+    * */
+    {
+        void onCallback(UserProfileObject userObject);
+    }
+
+    /**
+     * Edits FirebaseAuth email and Firebase database email/phone number of user
+     *
+     * @param inputEmail New email to be written
+     * @param inputPhone New phone number to be written
+     */
+    public static void changeAuthInfo(EditText inputEmail, EditText inputPhone, String documentID)
+    {
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+
+        user.updateEmail(inputEmail.getText().toString())
+                .addOnCompleteListener(new OnCompleteListener<Void>()
+                {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task)
+                    {
+                        if (task.isSuccessful())
+                        {
+                            /* Successful profile edit*/
+                            editEmail(documentID,
+                                    inputEmail.getText().toString());
+                            editPhone(documentID,
+                                    inputPhone.getText().toString());
+                            Log.d(TAG, "User info updated.");
+                        }
+                        else
+                        {
+                            try
+                            {
+                                /* Tries to match errorCode to EditText error */
+                                String errorCode = "";
+                                errorCode = ((FirebaseAuthException) task.getException()).getErrorCode();
+
+                                switch (errorCode)
+                                {
+                                    case "ERROR_INVALID_EMAIL":
+                                        /* Set Email EditText error code to check validity */
+                                        EditTextValidator.invalidEmail(inputEmail);
+                                        break;
+
+                                    case "ERROR_EMAIL_ALREADY_IN_USE":
+                                        /* Set Email EditText error code to email taken */
+                                        EditTextValidator.emailTaken(inputEmail);
+                                        break;
+
+                                    default:
+                                        /* Unexpected Error code*/
+                                        inputEmail.setError(errorCode);
+                                        throw new Exception("Unexpected Firebase Error Code"
+                                                + "inside click listener.");
+                                }
+                            }
+                            catch (Exception e)
```

Comment:
I put this comment on the other PR but it got closed: 

Is this catch only triggered by the thrown exception in the default case? If so I would remove the try/catch because firebase error codes might not necessarily be a programming error (for example, what if we lose connection to firebase?)

Instead I would do:

default:
Log.d(Tag, "Unexpected Firebase Error code: " + errorCode);

This code might not be syntactically correct, but you get the idea.

Reply:
this catch is the message "This operation is sensitive and requires re log before performing"

null crash: cannot be cast to (FirebaseAuthException) task.getException()).getErrorCode()